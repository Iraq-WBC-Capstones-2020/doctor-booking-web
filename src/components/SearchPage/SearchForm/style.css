.searchForm {
  margin-top: 100px;
  margin-bottom: 148px;
}

.searchHead {
  margin-bottom: 79.56px;
}

.searchHead h1 {
  font-weight: normal;
  font-size: 64px;
  line-height: 75px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.searchHead img {
  margin-left: 30px;
}

.form-row input {
  height: 51px;
}

.form-row button {
  color: white;
  background-color: #026fc2;
}

.myButton {
  font-size: 18px;
  height: 51px;
  width: 100%;
  color: white;
  border: none;
  border-radius: 5px;
  background: #026fc2;
}

.myButton svg {
  margin-left: 5px;
}
/*# sourceMappingURL=style.css.map */