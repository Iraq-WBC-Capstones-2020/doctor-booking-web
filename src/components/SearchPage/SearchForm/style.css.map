{
    "version": 3,
    "mappings": "AACA,AAAA,WAAW,CAAC;EACV,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,KAAK;CACrB;;AACD,AAAA,WAAW,CAAC;EACV,aAAa,EAAE,OAAO;CAYvB;;AAbD,AAEE,WAFS,CAET,EAAE,CAAC;EACD,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,OAAO,EAAE,IAAI;EACb,KAAK,EAAE,OAAO;CACf;;AARH,AAUE,WAVS,CAUT,GAAG,CAAC;EACF,WAAW,EAAE,IAAI;CAClB;;AAGH,AACE,SADO,CACP,KAAK,EADP,SAAS,CACA,MAAM,CAAC;EACZ,aAAa,EAAC,IAAK;EACnB,MAAM,EAAE,IAAK;CACd;;AAJH,AAME,SANO,CAMP,MAAM,CAAC;EACL,KAAK,EAAE,KAAK;EACZ,gBAAgB,EC3BV,OAAO;CD4Bd;;AAGH,AAAA,SAAS,CAAC;EACR,SAAS,ECjCH,IAAI;EDkCV,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,aAAa,EAAE,GAAG;EAClB,UAAU,ECtCF,OAAO;CD2ChB;;AAZD,AASE,SATO,CASP,GAAG,CAAC;EACF,WAAW,EAAE,GAAG;CACjB",
    "sources": [
        "style.scss",
        "../_variable.scss"
    ],
    "names": [],
    "file": "style.css"
}