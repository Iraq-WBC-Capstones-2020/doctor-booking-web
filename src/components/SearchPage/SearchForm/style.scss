@import '../variable';
.searchForm{
    margin-bottom: 148px;
}
.searchHead{
    margin-bottom: 79.56px;
    h1{
    font-weight: normal;
    font-size: 64px;
    line-height: 75px;
    display: flex;
    color: #000000;
    }
    

    img{
        margin-left: 30px;
    };

}

.form-row{
    input{
        height: 51px;
    }

    button{
        color: white;
        background-color: $theBlue;
    }
}

.myButton{
    font-size: $pFont;
    height: 51px;
    width: 179px;
    color: white;
    border: none;
    border-radius: 5px;
    background:$theBlue;
    
    svg{
        margin-left: 5px;
    }
}