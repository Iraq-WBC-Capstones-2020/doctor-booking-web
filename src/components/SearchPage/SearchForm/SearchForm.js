import React, { useState, useEffect, useContext } from 'react';
import { Container, Row, Form, Col } from 'react-bootstrap';
import SearchIcon from './images/Search.svg';
import { db } from '../../../Firebase';
import { DoctorContext, ACTIONS } from '../../../DoctorContext';

import './style.css';
function SearchForm({ setFilterDoctors, input, setInput }) {
  const [spec, setSpec] = useState('');

  const [state, dispatch] = useContext(DoctorContext);

  const speciality = [
    'All',
    'Abdominal radiology',
    'Breast imaging',
    'Cardiothoracic radiology',
    'Cardiovascular radiology',
    'Chest radiology',
    'Emergency radiology',
    'Endovascular surgical neuroradiology',
    'Gastrointestinal radiology',
    'Genitourinary radiology',
    'Head and neck radiology',
    'Pulmonary disease',
    'Pulmonary disease',
    'Sleep medicine',
    'Sports medicine',
    'Transplant hepatology',
  ];

  //grab date from firestore and put in global state
  useEffect(() => {
    db.collection('doctors').onSnapshot((snapshot) => {
      dispatch({
        type: ACTIONS.GET_DOCTORS,
        payload: { doctors: snapshot.docs.map((doc) => doc.data()) },
      });
    });
  }, []);

  useEffect(() => {
    setFilterDoctors(
      state.doctors.filter((doctor) => {
        return doctor.name.toLowerCase().includes(input.toLowerCase());
      })
    );
  }, [input]);

  function handleSubmit(e) {
    e.preventDefault();
    setFilterDoctors(
      state.doctors.filter((doctor) => {
        return (
          doctor.speciality === spec &&
          doctor.name.toLowerCase().includes(input.toLowerCase())
        );
      })
    );
    setInput('');
    setSpec('');
  }

  return (
    <div className="searchForm">
      <Container>
        <Row className="searchHead text-center">
          <Col lg={12}>
            <h1>Search for your Doctor</h1>
            <img src={SearchIcon} alt="search icon" />
          </Col>
        </Row>
        <Form.Row
          className="justify-content-center"
          as="form"
          onSubmit={handleSubmit}
        >
          <Col className="myCol" sm={4}>
            <Form.Control
              className="inputForm"
              value={input}
              onChange={(e) => setInput(e.target.value)}
              placeholder="Doctor name"
            />
          </Col>
          <Col className="myCol" sm={4}>
            <Form.Group
              className="inputForm"
              controlId="exampleForm.SelectCustom"
            >
              <Form.Control
                className="inputForm"
                as="select"
                value={spec}
                onChange={(e) => setSpec(e.target.value)}
                custom
              >
                {speciality.map((spec, index) => (
                  <option key={index}>{spec}</option>
                ))}
              </Form.Control>
            </Form.Group>
          </Col>
          <Col sm={2}>
            <button className="myButton" type="submit">
              Search
              <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-search"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  fillRule="evenodd"
                  d="M10.442 10.442a1 1 0 0 1 1.415 0l3.85 3.85a1 1 0 0 1-1.414 1.415l-3.85-3.85a1 1 0 0 1 0-1.415z"
                />
                <path
                  fillRule="evenodd"
                  d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zM13 6.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z"
                />
              </svg>
            </button>
          </Col>
        </Form.Row>
      </Container>
    </div>
  );
}

export default SearchForm;
