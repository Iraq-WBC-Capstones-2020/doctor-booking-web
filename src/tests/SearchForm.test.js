import React from 'react';
import { render } from '@testing-library/react';
import SearchForm from '../components/SearchPage/SearchForm/SearchForm';
import renderer from 'react-test-renderer';
import DoctorInfoProvider from '../DoctorContext';
it('render1', () => {
  render(
    <DoctorInfoProvider>
      <SearchForm />
    </DoctorInfoProvider>
  );
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DoctorInfoProvider>
        <SearchForm />
      </DoctorInfoProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
