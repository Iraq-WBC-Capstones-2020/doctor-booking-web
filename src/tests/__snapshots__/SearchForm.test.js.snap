// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="searchForm"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="searchHead"
      >
        <div
          className="col-lg-12"
        >
          <h1>
            Search for your doctor
            <img
              src="Search.png"
            />
          </h1>
        </div>
      </div>
    </div>
    <div
      className="justify-content-center form-row"
    >
      <div
        className="myCol col-sm-4"
      >
        <input
          className="inputForm form-control"
          placeholder="Doctor name"
        />
      </div>
      <div
        className="myCol col-sm-4"
      >
        <div
          className="inputForm form-group"
        >
          <select
            className="inputForm custom-select"
            id="exampleForm.SelectCustom"
          >
            <option>
              Abdominal radiology
            </option>
            <option>
              Breast imaging
            </option>
            <option>
              Cardiothoracic radiology
            </option>
            <option>
              Cardiovascular radiology
            </option>
            <option>
              Chest radiology
            </option>
            <option>
              Emergency radiology
            </option>
            <option>
              Endovascular surgical neuroradiology
            </option>
            <option>
              Gastrointestinal radiology
            </option>
            <option>
              Genitourinary radiology
            </option>
            <option>
              Head and neck radiology
            </option>
            <option>
              Pulmonary disease
            </option>
            <option>
              Pulmonary disease
            </option>
            <option>
              Sleep medicine
            </option>
            <option>
              Sports medicine
            </option>
            <option>
              Transplant hepatology
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-sm-2"
      >
        <button
          as="input"
          className="myButton"
          value="Submit"
        >
          Search
          <svg
            className="bi bi-search"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.442 10.442a1 1 0 0 1 1.415 0l3.85 3.85a1 1 0 0 1-1.414 1.415l-3.85-3.85a1 1 0 0 1 0-1.415z"
              fillRule="evenodd"
            />
            <path
              d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zM13 6.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
